src/plugins/data/common/search/aggs/agg_types_registry.ts

```diff
@@ -60,11 +76,15 @@ export class AggTypesRegistry {
     };
 
     return {
-      get: (name: string) =>
-        getInitializedFromCache<BucketAggType<any> | MetricAggType<any>>(
-          name,
-          this.bucketAggs.get(name) || this.metricAggs.get(name)
-        ),
+      get: (name: string) => {
+        const agg = this.bucketAggs.get(name) || this.metricAggs.get(name) || this.legacyAggs.get(name);
```

Comment:
map.get can return undefined. Updating method to handle this case. This required some `!` in tests